hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/io/TestHoodieTimelineArchiveLog.java

```diff
@@ -243,6 +243,36 @@ public void testArchiveCommitSavepointNoHole() throws Exception {
         "Archived commits should always be safe");
   }
 
+  @Test
+  public void testPendingClusteringWillBlockArchival() throws Exception {
+    init();
+    HoodieWriteConfig cfg = HoodieWriteConfig.newBuilder().withPath(basePath)
+        .withSchema(HoodieTestDataGenerator.TRIP_EXAMPLE_SCHEMA).withParallelism(2, 2).forTable("test-trip-table")
+        .withCompactionConfig(HoodieCompactionConfig.newBuilder().retainCommits(1).archiveCommitsWith(2, 5).build())
+        .withFileSystemViewConfig(FileSystemViewStorageConfig.newBuilder()
+            .withRemoteServerPort(timelineServicePort).build())
+        .withMetadataConfig(HoodieMetadataConfig.newBuilder().enable(false).build())
```

Comment:
can you do `@ParameterizedTest` of true/false for metadata table

Reply:
Changed.